File: fixtures/scale/java/large/src/cryptobench/ciphers/speck/SpeckSuite.java
package cryptobench.ciphers.speck;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class SpeckSuite implements CipherSuite {
    @Override
    public String name() {
        return "speck";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new SpeckCipher(SpeckKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : SpeckVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/speck/SpeckSuite.scala
package cryptobench.ciphers.speck

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class SpeckSuite extends CipherSuite {
  override def name: String = "speck"

  override def run(): SuiteResult = {
    val cipher: Cipher = new SpeckCipher(SpeckKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- SpeckVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}
